test/user.test.js

```diff
@@ -1858,77 +1785,139 @@ describe('User', function() {
-            let actualVerifyHref;
-            return user.verify({
-              type: 'email',
-              to: user.email,
-              from: 'noreply@myapp.org',
-              redirect: '#/some-path?a=1&b=2',
-              templateFn: (options, cb) => {
-                actualVerifyHref = options.verifyHref;
-                cb(null, 'dummy body');
-              },
-            })
-            .then(() => actualVerifyHref);
-          })
+        let actualVerifyHref;
+
+        Object.assign(verifyOptions, {
+          redirect: '#/some-path?a=1&b=2',
+          templateFn: (verifyOptions, cb) => {
+            actualVerifyHref = verifyOptions.verifyHref;
+            cb(null, 'dummy body');
+          },
+        });
+
+        return user.verify(verifyOptions)
+          .then(() => actualVerifyHref)
           .then(verifyHref => {
             var parsed = url.parse(verifyHref, true);
             expect(parsed.query.redirect, 'redirect query')
               .to.equal('#/some-path?a=1&b=2');
           });
       });
 
-      it('verify that options.templateFn receives options.verificationToken', function() {
-        return User.create({email: 'test1@example.com', password: 'pass'})
-          .then(user => {
-            let actualVerificationToken;
-            return user.verify({
-              type: 'email',
-              to: user.email,
-              from: 'noreply@myapp.org',
-              redirect: '#/some-path?a=1&b=2',
-              templateFn: (options, cb) => {
-                actualVerificationToken = options.verificationToken;
-                cb(null, 'dummy body');
-              },
-            })
-            .then(() => actualVerificationToken);
-          })
+      it('verify that verifyOptions.templateFn receives ' +
+        'verifyOptions.verificationToken', function() {
+        let actualVerificationToken;
+
+        Object.assign(verifyOptions, {
+          redirect: '#/some-path?a=1&b=2',
+          templateFn: (verifyOptions, cb) => {
+            actualVerificationToken = verifyOptions.verificationToken;
+            cb(null, 'dummy body');
+          },
+        });
+
+        return user.verify(verifyOptions)
+          .then(() => actualVerificationToken)
           .then(token => {
             expect(token).to.exist();
           });
       });
+
+      it('forwards the "options" argument to user.save() ' +
+        'when adding verification token', function() {
+        let onBeforeSaveCtx;
+
+        // before save operation hook to capture remote ctx when saving
+        // verification token in user instance
+        User.observe('before save', function(ctx, next) {
+          onBeforeSaveCtx = ctx;
+          next();
+        });
+
+        return user.verify(verifyOptions, ctxOptions)
+          .then(() => {
+            // not checking equality since other properties are added by user.save()
+            expect(onBeforeSaveCtx.options).to.contain({testFlag: true});
```

Comment:
If the hook was not called, then this would throw a TypeError because `onBeforeSaveCtx` is not defined. A possible solution: initalise `onBeforeSaveCtx` with an empty object.